firecrown/ccl/systematics/lss.py

```diff
@@ -46,3 +47,49 @@ def apply(self, cosmo, params, source):
         pref *= ccl.growth_factor(
                 cosmo, 1.0 / (1.0 + source.z_)) ** params[self.alphag]
         source.bias_ *= pref
+
+
+class MagnificationBias(Systematic):
+    """Magnification bias systematic.
+
+    This systematic adds a magnification bias model for galaxy number contrast
+    following Joachimi & Bridle (2010), arXiv:0911.2454.
+
+    Parameters
+    ----------
+    r_lim : str
+        The name of the limiting magnitude in r band filter.
+    Sig_c, eta, z_c, z_m : str
+        The name of the fitting parameters in Joachimi & Bridle (2010) equation
+    (C.1).
+
+    Methods
+    -------
+    apply : appaly the systematic to a source
+    """
+    def __init__(self, r_lim, Sig_c, eta, z_c, z_m):
+        self.r_lim = r_lim
+        self.Sig_c = Sig_c
+        self.eta = eta
+        self.z_c = z_c
+        self.z_m = z_m
+
+    def apply(self, cosmo, params, source):
+        """Apply a magnification bias systematic.
+
+        Parameters
+        ----------
+        cosmo : pyccl.Cosmology
+            A pyccl.Cosmology object.
+        params : dict
+            A dictionary mapping parameter names to their current values.
+        source : a source object
+            The source to which apply the shear bias.
+        """
+
+        z_bar = params[self.z_c] + params[self.z_m] * (params[self.r_lim] - 24)
+        z = source.z_
+        s = (
+            params[self.eta] / params[self.r_lim] - 3 * params[self.z_m] /
+            z_bar + 1.5 * params[self.z_m] * np.power(z / z_bar, 1.5) / z_bar)
+        source.mag_bias_ = s / np.log(10)
```

Comment:
We should be replacing this attribute. Instead we should be using something like `*=` here. This will let people declare an overall amplitude that can be a parameter as well.